test/rules/padding-between-test-function.js

```diff
@@ -0,0 +1,49 @@
+
+'use strict';
+
+var RuleTester = require('eslint').RuleTester,
+    rules = require('../../').rules,
+    ruleTester = new RuleTester(),
+    ALWAYS_MESSAGE = 'Test functions must be padded by blank lines.',
+    NEVER_MESSAGE = 'Test functions must not be padded by blank lines.';
+
+ruleTester.run('padding-between-tests', rules['padding-between-test-functions'], {
+
+    valid: [
+        {
+            code: 'describe(function(){\n\nit("", function () {})\n\n})'
+        },
+        {
+            code: 'describe(function(){\n\nit("", function () {})\n\n})',
+            options: [ 'always' ]
+        },
+        {
+            code: 'describe(function(){\n\nit.only("", function () {})\n\n})',
+            options: [ 'always' ]
+        },
+        {
+            code: 'describe(function(){\nit("", function () {})\n})',
+            options: [ 'never' ]
+        },
+        {
+            code: 'describe(function(){\nit.only("", function () {})\n})',
+            options: [ 'never' ]
+        }
+    ],
+
+    invalid: [
+        {
+            code: 'describe(function(){\n\nit("", function () {})\n\n})',
+            options: [ 'never' ],
+            errors: [ { message: NEVER_MESSAGE }, { message: NEVER_MESSAGE } ],
+            output: 'describe(function(){\nit("", function () {})\n})'
+        },
+        {
+            code: 'describe(function(){\nit("", function () {})\n})',
+            options: [ 'always' ],
+            errors: [ { message: ALWAYS_MESSAGE }, { message: ALWAYS_MESSAGE } ],
+            output: 'describe(function(){\n\nit("", function () {})\n\n})'
+        }
+    ]
```

Comment:
There is currently also no test case for multiple empty lines between test functions.

Reply:
Hmmm.... yes, multiple empty lines.  The code doesn't handle this case currently.  While it is possible to add an option for, say, number of spaces allowed before/after test functions - I believe this might trample on another eslint rule -- `no-multiple-empty-lines`.  I think it makes the most sense to let the user decide what is best here by setting that rule.

That said, I have noticed a few other problems in adding more tests like you suggested -- the trailing newline doesn't account for semicolons and will dutifully add a lineline between the end of statement `)` and the semicolon.  Working on this now.

Reply:
Huh? Isn’t this condition `second.loc.start.line - first.loc.end.line >= 2;` allowing more than one empty line? IMHO this is also the right thing to do, I just wanted to see a test that the rule also works correctly when the padding is more than one empty line.